### ISSUE TITLE
M2.1.3 : Shipped version of phpunit does not support PHP7 return types

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#7899]
If you want to write tests for your modules and you used return types, you cannot mock these. My best guess is that this is because Magento 2 ships with an old version of PHPUnit (4.1.0). PHPUnit is currently at version 5.7 and the last stable version of PHPUnit 4 is 4.8.

### Preconditions

1. Magento 2.1.3
2. PHPUnit 4.1.0 (ships with Magento 2)

### Steps to reproduce

1.  Create a class or interface that has a return type.
2. Try to mock it

Example:

```php
interface FooInterface
{
   public function getBar() : string;
}
```

```php
$fooMock = $this->getMock(\FooInterface::class);
```

### Expected result

PHPUnit should just create a mock of the interface. 

### Actual result

You get a PHPUnit error that the declaration is not compatible:

```
PHP Fatal error:  
Declaration of Mock_FooInterface_5b08a4e7::getBar() must be compatible with Vendor\Module\Api\FooInterface::getBar(): string
```

This makes sense of course. So why does Magento still uses an old version of PHPUnit that is not PHP7-compatible?

**Comment by TomashKhamlai** [comment]
Thanks for reporting this issue! We've created internal ticket MAGETWO-70104 to address this issue.

**Comment by TomashKhamlai** [comment]
@kanduvisla if you want to mock functions that have return types, you can try this little instruction.
Keep in mind:
1) DO NOT use it on a production server
2) This is not a well tested official solution, but it is worth to try it
3) You are doing this at your own risk! 

Open composer.json in a text-editor. Find the line in the code where 'phpunit' is located, and you will see something like:
```
"require-dev": {
        "phpunit/phpunit": "4.1.0",
```
replace "4.1.0" to "5.7.20"

Run command from CLI:
`composer update`

Now, all that you need is to change deprecated methods, for example, change getMock() to createMock().

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
The issue is already fixed in develop branch, 2.2.0
